test/connection_switching_test.rb

```diff
@@ -616,17 +616,6 @@ class ShardedModel < ActiveRecord::Base
           assert_equal ["slave_name", "slave_name2"], Account.pluck(:name)
         end
 
-        it "supports implicit joins" do
-          accounts = Account.includes(:tickets)
-          accounts = accounts.references(:tickets) if ActiveRecord::VERSION::MAJOR >= 4
-          assert_equal ["slave_name", "slave_name2"], accounts.order('tickets.id').map(&:name).sort
-        end
-
-        it "supports joins" do
-          accounts = Account.joins('LEFT OUTER JOIN tickets ON tickets.account_id = accounts.id').map(&:name).sort
-          assert_equal ["slave_name", "slave_name2"], accounts
-        end
```

Comment:
need a replacement test for this that is not cross-shard ?

Reply:
Done. We should probably test joins between sharded tables as well as between unsharded tables?